### PR TITLE
[MGMT-2317] Added boot discovery iso over ipxe from grub docs

### PR DESCRIPTION
**Pull request by mvazquezc**
Adds documentation around how to boot a discovery iso from an iPXE server using GRUB.

### CODE REVIEW

File: docs/user-guide/boot-discovery-iso-over-ipxe-from-grub.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+    ~~~sh
+    export IPXE_DIR=/tmp/ipxe/ai
+    mkdir -p ${IPXE_DIR}
+    ~~~
+2. Run this container image for extracting the required files for PXE booting from the discovery image iso file
+
+    ~~~sh
+    # Container code: https://github.com/ohadlevy/ai-ipxe
+    podman run -e BASE_URL=http://devscripts2ipv6.e2e.bos.redhat.com:8080/ipxe/ -e ISO_URL=http://10.19.115.219:6008/api/assisted-install/v1/clusters/1c907764-6354-409e-9a2e-93582497545a/downloads/image -v /tmp/ipxe/ai:/data:Z --net=host -it --rm quay.io/ohadlevy/ai-ipxe:latest
```
Comment (avishayt) [inline docs/user-guide/boot-discovery-iso-over-ipxe-from-grub.md:19]:
```suggestion
    podman run -e BASE_URL=http://devscripts2ipv6.e2e.bos.redhat.com:8080/ipxe/ -e ISO_URL=<REPLACE WITH ISO UR> -v /tmp/ipxe/ai:/data:Z --net=host -it --rm quay.io/ohadlevy/ai-ipxe:latest
```

File: docs/user-guide/boot-discovery-iso-over-ipxe-from-grub.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    ~~~
+3. Run an nginx container for exposing the required files over http
+
+    ~~~sh
+    podman run  -v ${IPXE_DIR}:/app:ro,Z -p 8080:8080 -d --rm bitnami/nginx:latest
```
Comment (avishayt) [inline docs/user-guide/boot-discovery-iso-over-ipxe-from-grub.md:24]:
```suggestion
    podman run -v ${IPXE_DIR}:/app:ro,Z -p 8080:8080 -d --rm bitnami/nginx:latest
```

File: docs/user-guide/boot-discovery-iso-over-ipxe-from-grub.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    curl http://devscripts2ipv6.e2e.bos.redhat.com:8080/ipxe/ipxe
+    ~~~
+
+    ~~~sh
+    #!ipxe                                                                                                                                                                                    
+    set live_url http://devscripts2ipv6.e2e.bos.redhat.com:8080/ipxe/
+    kernel ${live_url}/vmlinuz ignition.config.url=${live_url}/config.ign coreos.live.rootfs_url=${live_url}/rootfs.img random.trust_cpu=on rd.luks.options=discard ignition.firstboot ignition.platform.id=metal console=tty0 console=ttyS0,115200n8 coreos.inst.persistent-kargs="console=tty0 console=ttyS1,115200n8"
+    initrd ${live_url}/initrd.img
+    boot   
+    ~~~
```
Comment (avishayt) [inline docs/user-guide/boot-discovery-iso-over-ipxe-from-grub.md:38]: Where does this go?

File: docs/user-guide/boot-discovery-iso-over-ipxe-from-grub.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+Now that we have the PXE files available over http, we are going to configure GRUB for booting from that server
+
+1. Login into the server you want to pxe boot as root
+2. Get the iPXE boot module
+
+    ~~~sh
+    curl http://boot.ipxe.org/ipxe.lkrn -o /boot/ipxe.lkrn
+    ~~~
+3. Create a iPXE script for running the DHCP request and for chainload the pxe files from the server configured previously
```
Comment (avishayt) [inline docs/user-guide/boot-discovery-iso-over-ipxe-from-grub.md:50]:
```suggestion
3. Create an iPXE script for running the DHCP request and for chainload the PXE files from the server configured previously
```

File: docs/user-guide/boot-discovery-iso-over-ipxe-from-grub.md (hunk 3 above)
Comment (mvazquezc) [reply docs/user-guide/boot-discovery-iso-over-ipxe-from-grub.md:38]: It is the output from the curl command.

**Comment by droslean** [comment]
For the record: I tried to follow those steps but without any success. I am not sure if there are any other dependencies but it worth checking and include those in this document.

**Comment by droslean** [comment]
It seems that booting from iPXE really depends on the hardware. I will continue to investigate.

**Comment by flaper87** [comment]
Is this PR still needed ? O can we close it? :)

**Review by flaper87 — CHANGES_REQUESTED** [review]
@mvazquezc thanks for writing these docs. 

I was browsing the docs we have and noticed that there is already some documentation for iPXE deployment. You can find it here: https://github.com/openshift/assisted-service/blob/master/docs/user-guide/deploy-on-bare-metal.md#ipxe

* Were you aware of it? 
* How does the existing one differs from the one you wrote?
* Would it be possible to merge them? Maybe link this one in the `Manual` section of the existing one?

**Comment by mvazquezc** [comment]
> @mvazquezc thanks for writing these docs.
> 
> I was browsing the docs we have and noticed that there is already some documentation for iPXE deployment. You can find it here: https://github.com/openshift/assisted-service/blob/master/docs/user-guide/deploy-on-bare-metal.md#ipxe
> 
> * Were you aware of it?

When this PR was created I believe that doc didn't exist.

> * How does the existing one differs from the one you wrote?

It uses GRUB to boot into iPXE, it's useful for environments where you don't have access to the BMC, like cloud providers.

> * Would it be possible to merge them? Maybe link this one in the `Manual` section of the existing one?

Works for me.

**Comment by flaper87** [comment]
/close

Closing for inactivity. Reopen when ready to update/merge

**Pull request closed without merging**